xbmc/utils/StreamDetails.cpp

```diff
@@ -398,6 +402,16 @@ int CStreamDetails::GetVideoHeight(int idx) const
     return 0;
 }
 
+std::string CStreamDetails::GetVideoHdrType( int idx) const
+{
+  const CStreamDetailVideo* item =
+      static_cast<const CStreamDetailVideo*>(GetNthStream(CStreamDetail::VIDEO, idx));
```

Comment:
Seems, you are doing an upcast here and looking at the `if (item)` on the next line, should this rather be a `dynamic_cast`?

Reply:
Why? We don't need a runtime check here because we know what the object is.  Also, it's consistent with all the other getters. I'm happy to change it though if you'd rather, you know much more about this stuff than me.

Reply:
Technically `static_cast` is fine here. You call `GetNthStream()` and specify that it's a video stream, so the return value must be a `CStreamDetailVideo` or null. Otherwise, it's a programming error. `GetNthStream()` essentially does the work of upcasting, so `static_cast` is sufficient here.

However, if we want to do defensive programming (always a good idea) then dynamic_cast would be a harmless improvement.

Reply:
Personally, I would never rely on what the **implementation** of `GetNthStream` does. This can change at any time as it is not part of the contract, the **interface**. And then your code breaks. The contract says `const CStreamDetail* GetNthStream(CStreamDetail::StreamType type, int idx) const`. So, it is technically not guaranteed that you will get a `const CStreamDetailVideo*`.

Reply:
@ksooo, @garbear  What do you want to do here?  As ksooo's comment applies to all the other existing getters too I suggest that in the interests of code consistency that it remains as `static_cast` for the purposes of this PR, with a follow-up PR to update all the getters in `CStreamDetails.cpp` to use `dynamic_cast` - that will also reformat the code with clang in line with the current guidelines.
If that isn't acceptable, please suggest an alternative that will be.

Reply:
Follow-up PR is fine for me.